pyanaconda/modules/payloads/source/rpm_ostree_container/utils.py

```diff
@@ -0,0 +1,42 @@
+#
+# The RPM OSTree utilities.
+#
+# Copyright (C) 2023 Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+
+def containers_needs_network(url):
+    """Detect if the container registry URL needs network.
+
+    :return: True or False
+    """
+    if not url:
+        return False
+    # Only docker:// and registry:// (could be omitted) should be used for remote installations
+    # however, you can use 'localhost' in these situations to connect local resources.
+    if url.startswith("docker://localhost") or \
+       url.startswith("registry://localhost") or \
+       url.startswith("localhost"):
+        return False
+
+    # Not connecting localhost (above check) anything else in docker:// or registry:// should
+    # require network.
+    if url.startswith("docker://") or \
+       url.startswith("registry://") or \
+       "://" not in url:
+        return True
+
+    return False
```

Comment:
@cgwalters What do you think about this solution for network detection?

Reply:
Sure, seems fine to start.  It'd be better probably to have this be a shared library function somewhere, we could even do something like `ostree container url-requires-network <imgref>` or perhaps `skopeo image-requires-network <imgref>`.

But I'm not really concerned about this to start, I think the 99% cases are:

- netinst ISO pulling from a remote registry
- "Full" ISO pulling from embedded content i.e. `oci:/path/to/embedded/image` as is done for desktop installers today, and is also how the Fedora CoreOS and derivative Live ISO works.

-

Reply:
Great idea about the shared version. Please ping us in case it's implemented.

I think you are correct but I rather play safe in Anaconda about these.